src/Http/Middleware/RateLimit.php

```diff
@@ -64,7 +64,8 @@ public function handle($request, Closure $next)
         $this->handler->rateLimitRequest($request, $route->getRateLimit(), $route->getRateLimitExpiration());
 
         if ($this->handler->exceededRateLimit()) {
-            throw new HttpException(403, 'You have exceeded your rate limit.', null, $this->getHeaders());
+            $retryAfter = $this->getHeaders()['X-RateLimit-Reset'] - time();
+            throw new TooManyRequestsHttpException($retryAfter, 'You have exceeded your rate limit.');
```

Comment:
I agree with @lucasmichot. Would be nice if we could throw the actual named exception, but seems Symfony doesn't want us to be able to set custom headers on that exception.